src/persist-client/src/operators/shard_source.rs

```diff
@@ -303,6 +306,10 @@ where
         // will block when there is no data yet available in the shard.
         cap_set.downgrade(as_of.clone());
 
+        // Wait for the start signal before ingesting any data. Do this after downgrading the
```

Comment:
I'd imagined this happening even before we register the reader (which starts the background heartbeat loop, etc). Is the initial downgrade stuff relevant when the whole dataflow is suspended?

Reply:
Probably doesn't matter much where we put this. persist_sink won't be able to make progress, but we don't want to hydrate the dataflow anyway yet, so that's fine.

I thought it would be a good idea to make sure we already get the reader lease before we started the waiting, to be sure we have it once the suspension is lifted. It shouldn't make a difference because the compute controller should hold back the read frontier appropriately, but there can always be bugs. On the other hand, you could argue that this would be a great opportunity to find them, and I'd agree! But I suspect you have a different reason for not wanting to start the heartbeat task early?

Reply:
> But I suspect you have a different reason for not wanting to start the heartbeat task early?

Not a good one! Just running the exercise of "how close can we make suspended to unrendered?"

> I thought it would be a good idea to make sure we already get the reader lease before we started the waiting, to be sure we have it once the suspension is lifted. It shouldn't make a difference because the compute controller should hold back the read frontier appropriately, but there can always be bugs. On the other hand, you could argue that this would be a great opportunity to find them, and I'd agree!

:joy: